sonar-python-plugin/src/main/java/org/sonar/plugins/python/IPynbSensor.java

```diff
@@ -67,10 +77,38 @@ public void execute(SensorContext context) {
     if (pythonVersions.length != 0) {
       ProjectPythonVersion.setCurrentVersions(PythonVersionUtils.fromStringArray(pythonVersions));
     }
-    PythonScanner scanner = new PythonScanner(context, checks, fileLinesContextFactory, noSonarFilter, PythonParser.createIPythonParser(), indexer);
+    if (isInSonarLintRuntime(context)) {
+      PythonScanner scanner = new PythonScanner(context, checks, fileLinesContextFactory, noSonarFilter, PythonParser.createIPythonParser(), indexer);
+      scanner.execute(pythonFiles, context);
+    } else {
+      processNotebooksFiles(pythonFiles, context);
+    }
+  }
+
+  private void processNotebooksFiles(List<PythonInputFile> pythonFiles, SensorContext context) {
+    pythonFiles = parseNotebooks(pythonFiles);
+    // Disable caching for IPynb files for now
+    CacheContext cacheContext = CacheContextImpl.dummyCache();
```

Comment:
We should have that documented

Reply:
Done in https://sonarsource.atlassian.net/browse/SONARPY-2020